lib/compilers/cc65.ts

```diff
@@ -0,0 +1,127 @@
+import * as utils from '../utils';
+
+export class Cc65Compiler extends BaseCompiler {
+    static get key() {
+        return 'cc65';
+    }
+
+    constructor(compilerInfo, env) {
+        super(compilerInfo, env);
+
+        this.asm = new CC65AsmParser(this.compilerProps);
+        this.toolchainPath = path.resolve(path.dirname(compilerInfo.exe), '..');
+    }
+
+    override getSharedLibraryPathsAsArguments(libraries, libDownloadPath?) {
+        const libPathFlag = this.compiler.libpathFlag || '-L';
+
+        if (!libDownloadPath) {
+            libDownloadPath = '.';
+        }
+
+        return _.union(
+            [libPathFlag + libDownloadPath],
+            this.compiler.libPath.map(path => libPathFlag + path),
+            this.getSharedLibraryPaths(libraries).map(path => libPathFlag + path),
+        ) as string[];
+    }
+
+    override optionsForFilter(filters, outputFilename) {
+        if (filters.binary) {
+            return ['-g', '-o', this.filename(outputFilename)];
+        } else {
+            return ['-g', '-S', '-c', '-o', this.filename(outputFilename)];
+        }
+    }
+
+    override getCompilerEnvironmentVariables(compilerflags) {
+        const allOptions = (this.compiler.options + ' ' + compilerflags).trim();
+        return {...this.cmakeBaseEnv, CFLAGS: allOptions};
+    }
+
+    override async getCmakeBaseEnv() {
+        if (!this.compiler.exe) return {};
+
+        const env: Record<string, string> = {};
+
+        env.CC = this.compiler.exe;
+
+        if (this.toolchainPath) {
+            const ldPath = `${this.toolchainPath}/bin/ld65`;
+            const arPath = `${this.toolchainPath}/bin/ar65`;
+            const asPath = `${this.toolchainPath}/bin/as65`;
+
+            if (await utils.fileExists(ldPath)) env.LD = ldPath;
+            if (await utils.fileExists(arPath)) env.AR = arPath;
+            if (await utils.fileExists(asPath)) env.AS = asPath;
+        }
+
+        return env;
+    }
+
+    override async objdump(
+        outputFilename,
+        result: CompilationResult,
+        maxSize: number,
+        intelAsm,
+        demangle,
+        filters: ParseFilters,
+    ) {
+        const res = await super.objdump(outputFilename, result, maxSize, intelAsm, demangle, filters);
+
+        const dirPath = path.dirname(outputFilename);
+        const nesFile = path.join(dirPath, 'example.nes');
+        if (await utils.fileExists(nesFile)) {
+            const file_buffer = await fs.readFile(nesFile);
+            const binary_base64 = file_buffer.toString('base64');
+            result.jsnesrom = binary_base64;
+        }
+
+        return res;
+    }
+
+    override async doBuildstepAndAddToResult(result: CompilationResult, name, command, args, execParams) {
+        const stepResult = await super.doBuildstepAndAddToResult(result, name, command, args, execParams);
+        if (name === 'make') {
+            const mapFile = path.join(execParams.customCwd, '../map.txt');
```

Comment:
I'm probably not getting this properly but... in sandbox mode won't `customCwd` be like `/app` and so nothing can write to `../` of that?

Reply:
The cwd for ‘make’ is the ‘build’ folder, not sure if thats the brightest solution. But it will make it consistent for ppl to enable the memory map files also for non-cmake builds maybe. Not sure

Reply:
Ah but i see that i only read the map file if its a cmake project… so maybe the build folder should just suffice